src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -157,15 +155,37 @@ private void ExecuteCmdletWithAll()
             using (var client = this.CreateClient())
             {
                 var allWorkspaces = this.ExecuteCmdletWithAll((top, skip) => client.Workspaces.GetWorkspacesAsAdmin(expand: "users", filter: this.Filter, top: top, skip: skip));
+                var filteredWorkspaces = new List<Workspace>();
 
                 if (!string.IsNullOrEmpty(this.User))
                 {
                     allWorkspaces = allWorkspaces
-                        .Where(w => w.Users.Any(u => u.UserPrincipalName != null && u.UserPrincipalName.Equals(this.User, StringComparison.OrdinalIgnoreCase)))
+                        .Where(w => w.Users != null && w.Users.Any(u => u.UserPrincipalName != null && u.UserPrincipalName.Equals(this.User, StringComparison.OrdinalIgnoreCase)))
                         .ToList();
                 }
 
-                this.Logger.WriteObject(allWorkspaces, true);
+                if (this.Deleted.IsPresent && this.Orphaned.IsPresent)
```

Comment:
Why is this conditioned needed? Won't the seperate conditions for deleted and orphaned take care of this?

Reply:
removed the conditions since we no longer support -Deleted and -Orphaned together.